opentelemetry/proto/metrics/v1/metrics.proto

```diff
@@ -95,14 +95,140 @@ message InstrumentationLibraryMetrics {
+// aggregation function, the result of applying the DataPoint's
+// associated function of to one or more measurements.
+//
 // All DataPoint types have three common fields:
-// - Labels zero or more key-value pairs associated with the data point.
-// - StartTimeUnixNano MUST be set to the start of the interval when the data's
-//   type includes an AggregationTemporality. This field is not set otherwise.
-// - TimeUnixNano MUST be set to:
-//   - the moment when an aggregation is reported (independent of the
-//     aggregation temporality).
-//   - the instantaneous time of the event.
+// - Attributes includes key-value pairs associated with the data point
+// - TimeUnixNano is required, set to the end time of the aggregation
+// - StartTimeUnixNano is optional, but strongly encouraged for DataPoints
+//   having an AggregationTemporality field, as discussed below.
+//
+// Both TimeUnixNano and StartTimeUnixNano values are expressed as
+// UNIX Epoch time in nanoseconds since 00:00:00 UTC on 1 January 1970.
+//
+// # TimeUnixNano
+//
+// This field is required, having consistent interpretation across
+// DataPoint types.  TimeUnixNano is the moment corresponding to when
+// the data point's aggregate value was captured.
+//
+// Data points with/ the 0 value for TimeUnixNano SHOULD be rejected
+// by consumers.
+//
+// # StartTimeUnixNano
+//
+// StartTimeUnixNano in general allows detecting when a sequence of
+// observations is unbroken.  StartTimeUnixNano is set to the
+// beginning of an unbroken sequence of observations (all point kinds
+// except with delta aggregation temporality) or the end of an
+// unbroken sequence of observations (point kinds with delta
+// temporality).
+//
+// This field indicates to consumers the start time for points with
+// cumulative and delta AggregationTemporality, and it should be
+// included whenever possible to support correct rate calculation.
+// Although it may be omitted when the start time is truly unknown,
+// setting StartTimeUnixNano is strongly encouraged.
+//
+// There are two reasons that streams with AggregationTemporality are
+// reset: (1) a process restarts, and (2) the process "forgets" the stream
+// and then re-initializes it.  StartTimeUnixNano is critical for
+// calculating correct rates for Monotonic Sum, Histogram, and Summary
+// data points in both of these cases.
+//
+// Another use for this information which applies to all point kinds
+// is to prevent overlap when multiple writers accidentally write to a
+// single stream.  StartTimeUnixNano MAY be used for overlap detection
+// for all point kinds.
+//
+// ## StartTimeUnixNano: Sums and Histograms
+//
+// Since these two point kinds have AggregationTemporality, the use of
+// StartTimeUnixNano is strongly recommended to ensure correct rate
+// calculation and prevent overlap.
+//
+// ## StartTimeUnixNano: Gauges
+//
+// The use of StartTimeUnixNano is optional, recommended for overlap
```

Comment:
I'm glad you called this out. My thoughts:

1. I think calling start unix time nano optional for Gauges is fine.
2. The lack of aggregation temporarily (and by that I mean defining the window for the gauge) *or* clarity in guidance here is a bit concerning.   For now I think it's much simpler just to PICK one thing for this to represent rather than the verbage here which means you still can't interpret it if it would exist.
3. Does it make sense to define "Aggregation Temporality" (or something similar) on Gauge and *always* use it to interpret the meaning of start time?

My suggestion here is just to FORCE start time to be cumulative if it exists (so you *can* use it as a consumer) and have it be optional otherwise.

Reply:
For (2), see https://github.com/open-telemetry/opentelemetry-proto/issues/274. I would like to define "Instantaneous" temporality and say that Gauges have this temporality, implicitly, but it creates complications and there are other (also complicated) ways to model this.

Now that I think of it, I can't see any reason to prefer the delta interpretation I gave. A process can use its start time, costs almost nothing, and is correct.

Reply:
Yeah. If you say that start time for gauge is
- optional
- should be process start time (when it exists)

I'd be on board.  As long as it's clear how to interpret the data if it exists

Reply:
PTAL fd4d374.